job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/publisher/PublisherContext.groovy

```diff
@@ -153,6 +153,7 @@ class PublisherContext extends AbstractExtensibleContext {
             testResults(glob)
             keepLongStdio(junitContext.retainLongStdout)
             testDataPublishers(junitContext.testDataPublishersContext.testDataPublishers)
+            allowEmptyResults(junitContext.allowEmptyResults)
```

Comment:
This will break compatibility for everyone using an older version of the JUnit plugin. The new element should only be generated when a minimum version of the plugin is installed, e.g.

```groovy
if (jobManagement.isMinimumPluginVersionInstalled('junit', '1.10')) {
    allowEmptyResults(junitContext.allowEmptyResults)
}
```